homeassistant/components/unifiprotect/utils.py

```diff
@@ -19,3 +36,88 @@ def get_nested_attr(obj: Any, attr: str) -> Any:
         value = value.value
 
     return value
+
+
+async def profile_ws_messages(
+    hass: HomeAssistant,
```

Comment:
Both of these services could have quite a bit of their code offloaded to pyunifiprotect since they aren't doing anything home assistant specific for most of it

Reply:
I can move more of the profile code into pyunifiprotect (starting capture, waiting until complete, writing JSON file), but the generate_data one is already already all done in pyunifiprotect (except making the zip). A lot of the code is HA specific (generating the persistent notifications).